src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Patch/JsonPatchService.cs

```diff
@@ -82,7 +82,7 @@ private Resource GetPatchedJsonResource(FhirJsonNode node, JsonPatchDocument ope
             {
                 operations.ApplyTo(node.JsonObject);
             }
-            catch (JsonPatchException e)
+            catch (Exception e)
```

Comment:
What is the exception case that the general catch will cover that the JsonPatchException wont? I'm concerned about this catching and passing over an exception we would want logged.

Reply:
ArgmentNullException is one of them.
I also concern with widening catch net, but there is no way to validate JsonPatchDocument and we wrap only one string of code which we have not much control over.

I can add logging if you think it would be useful.

Reply:
My thought would be to have the specific exceptions (JsonPatch, ArgumentNull, etc) just throw the not valid exception, and have the catch all that still throws the request not valid exception but also logs the exception. That way we are aware of what is causing failures and we can either add it to the list of exceptions that don't log or fix it if it is a real bug.

Reply:
I revert widening, and add validation instead.